src/spbeam-halt/SPBEAMHaltSpell.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: © 2024 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity ^0.8.16;
+
+import {DssEmergencySpell} from "../DssEmergencySpell.sol";
+
+interface SPBEAMMomLike {
+    function halt(address) external;
+}
+
+interface SPBEAMLike {
+    function wards(address) external view returns (uint256);
+    function bad() external view returns (uint256);
+}
+
+/// @title SP-BEAM Halt Emergency Spell
+/// @notice Will disable the SPBEAM (Stability Parameter Bounded External Access Module)
+contract SPBEAMHaltSpell is DssEmergencySpell {
+    string public constant override description = "Emergency Spell | Halt SPBEAM";
+
+    SPBEAMMomLike public immutable spbeamMom = SPBEAMMomLike(_log.getAddress("SPBEAM_MOM"));
+    SPBEAMLike public immutable spbeam = SPBEAMLike(_log.getAddress("MCD_SP_BEAM"));
```

Comment:
As discussed, let's just modify the chainlog keys for this spell and keep it simple.
The chances of having different implementations for SPBEAM or other BEAMs are pretty slim, so the idea I proposed above would be overengineering.
```suggestion
    SPBEAMMomLike public immutable spbeamMom = SPBEAMMomLike(_log.getAddress("SP_BEAM_MOM"));
    SPBEAMLike public immutable spbeam = SPBEAMLike(_log.getAddress("SP_BEAM"));
```